### PR TITLE
rbac: apply the shadow_rules_stat_prefix to shadow stats in addition to dynamic metadata

### PR DESCRIPTION
**Pull request by yangminzhu**
In my original PR (https://github.com/envoyproxy/envoy/pull/15323), I intended to only apply the prefix to the dynamic metadata key emitted by the shadow_rules. However, looking at the current API description, the normal understanding is it should also apply to the normal `shadow_allowed` and `shadow_denied` stats. This PR adds the support for this.

We could add another field `shadow_rules_dynamic_metadata_prefix` in the API to allow configuring these separately but I'm not sure if people would ever need a different prefix for the same shadow rules. I can add it if it makes sense to anyone.

cc @mattklein123 who reviewed the last PR.

Commit Message: Apply the shadow_rules_stat_prefix to shadow stats in addition to dynamic metadata.
Additional Description: N/A
Risk Level: Low
Testing: Unit Tests
Docs Changes: Updated the RBAC filter documentation
Release Notes: N/A
Platform Specific Features: N/A

### CODE REVIEW

File: docs/root/configuration/listeners/network_filters/rbac_filter.rst [hunk 1]
```diff
@@ -23,6 +23,9 @@ Statistics
 
 The RBAC network filter outputs statistics in the *<stat_prefix>.rbac.* namespace.
 
+For the shadow rule statistics `shadow_allowed` and `shadow_denied`, the :ref:`shadow_rules_stat_prefix <envoy_v3_api_field_extensions.filters.http.rbac.v3.RBAC.shadow_rules_stat_prefix>`
```
Comment (mattklein123) [inline docs/root/configuration/listeners/network_filters/rbac_filter.rst:26]: Copy/paste issue here for the field? Should point to network filter? Same below?

**Review by mattklein123 — CHANGES_REQUESTED** [review]
Thanks this makes sense to me with small comment.

/wait

File: docs/root/configuration/listeners/network_filters/rbac_filter.rst (hunk 1 above)
Comment (yangminzhu) [reply docs/root/configuration/listeners/network_filters/rbac_filter.rst]: Ah, thanks for catching. Updated both.

**Review by mattklein123 — APPROVED** [review]
Thanks!

**Pull request merged**